### PR TITLE
swc: init at 0.91.19

### LINKED ISSUE #195677: nodePackages.swc-core = @swc/core
**Project description**

swc is the "speedy web compiler"
fast alternative to tsc, to compile typescript to javascript without type-checking

swc is mentioned in a different context in #113294

`npm install @swc/core` adds 150 MByte to node_modules
so it makes sense to factor-out swc to a global package

example use: [vscode](https://github.com/microsoft/vscode) (#26698)
on a 2x2GHz machine, swc is 10x faster than tsc = 2 minutes vs 20 minutes

@swc/cli is a lightweight wrapper (1 MByte or so)
but has @swc/core as peer-dependency, so npm will install both

**Metadata**

* homepage URL: https://swc.rs/
* source URL: https://github.com/swc-project/swc
* license: apl2
* platforms: all

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

Partially resolves #195677.

This adds the new `swc_cli` crate which is eventually aimed to wrap all swc commands. See more at https://github.com/swc-project/swc/discussions/3859. 

Is there any preference between `fetchCrate` vs `fetchFromGitHub`?

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/swc/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ lib
+, rustPlatform
+, fetchCrate
+}:
+
+rustPlatform.buildRustPackage {
+  name = "swc";
```
Comment (AndersonTorres) [inline pkgs/development/tools/swc/default.nix:7]: pname and version, please?

File: pkgs/development/tools/swc/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchCrate {
+    version = "0.91.11";
+    pname = "swc_cli";
+    sha256 = "sha256-vTmkuiMmvydLzMy+Qa0WST1mKSm2lFbG940VfKEeRI4=";
+  };
```
Comment (AndersonTorres) [inline pkgs/development/tools/swc/default.nix:12]:
```suggestion

  src = fetchCrate {
    pname = "swc_cli";
    version = "0.91.11";
    hash = "sha256-vTmkuiMmvydLzMy+Qa0WST1mKSm2lFbG940VfKEeRI4=";
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/swc/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/swc/default.nix:12]: and please inherit version

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/swc/default.nix (hunk 2 above)
Comment (dit7ya) [reply pkgs/development/tools/swc/default.nix:12]: `sha256` is a required argument for `fetchCrate`

File: pkgs/development/tools/swc/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+, fetchCrate
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "swc";
+  version = "0.91.12";
+
+  src = fetchCrate {
+    inherit version;
+    pname = "swc_cli";
```
Comment (AndersonTorres) [inline pkgs/development/tools/swc/default.nix:12]:
```suggestion
    pname = "swc_cli";
    inherit version;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**